### PR TITLE
chore: Add ability to configure maximum DB connection lifetime

### PR DESCRIPTION
**Pull request by mark9white**
Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [X] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [X] My organization is added to USERS.md.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -141,8 +141,9 @@ func (c PersistConfig) GetClusterName() string {
 }
 
 type ConnectionPool struct {
-	MaxIdleConns int `json:"maxIdleConns,omitempty"`
-	MaxOpenConns int `json:"maxOpenConns,omitempty"`
+	MaxIdleConns          int `json:"maxIdleConns,omitempty"`
+	MaxOpenConns          int `json:"maxOpenConns,omitempty"`
+	ConnMaxLifetimeMillis int `json:"connMaxLifetimeMillis,omitempty"`
```
Comment (simster7) [inline config/config.go:146]: I think just `ConnMaxLifetime` should be fine. Just document that it's in milliseconds in `docs/workflow-controller-configmap.yaml`

Comment (alexec) [reply config/config.go:146]: maybe use `time.Duration`?

Comment (mark9white) [reply config/config.go:146]: How about TTL? That's an alias for time.Duration that we have a json unmarshaller defined for, and this technically is a TTL.

**Comment by alexec** [comment]
LGHW

**Pull request merged**